layer1/chain/test/test_chain.mligo

```diff
@@ -0,0 +1,161 @@
+#include "../src/chain.mligo"
+#import "../../stdlib_ext/src/unit_test.mligo" "Unit"
+#import "../../stdlib_ext/src/result.mligo" "Stdlib_Result"
+
+
+let empty_chain : chain = 
+    { max_index = 0n 
+    ; batches = (Big_map.empty : (index, batch) big_map)
+    ; children = (Big_map.empty : (index, index list) big_map)
+    }
+
+let batch (parent:index) (level:nat) : batch = 
+        {  parent = parent
+        ;  level = level
+        ;  hash = 0x0101
+        ;  proposer = ("tz1gjaF81ZRRvdzjobyfVNsAeSC6PScjfQwN" : address)
+        }
+
+let _test_remove_one () = 
+    let first = batch 0n 0n in
+    let chain = 
+    { empty_chain with
+      max_index = 1n 
+    ; batches = Big_map.literal [(1n, first)]
+    ; children = (Big_map.empty : (index, index list) big_map)
+    } in
+    let new_chain = remove_batch (1n, chain) in
+    Unit.and_list 
+    [  Unit.assert_equals (None : batch option) (Big_map.find_opt 1n new_chain.batches) "batch should not be stored anymore"
+    ;  Unit.assert_equals 1n new_chain.max_index "max_index did not change"
+    ] 
+
+let _test_remove_father () =
+    let first = batch 0n 0n in
+    let second = batch 1n 0n in
+    let chain = 
+    { empty_chain with
+      max_index = 1n 
+    ; batches = Big_map.literal [(1n, first) ; (2n, second)]
+    ; children = Big_map.literal [(1n, [2n])]
+    } in
+    let new_chain = remove_batch (1n, chain) in
+    Unit.and_list 
+    [  Unit.assert_equals (None : batch option) (Big_map.find_opt 1n new_chain.batches) "batch 1 should not be stored anymore"
+    ;  Unit.assert_equals (None : batch option) (Big_map.find_opt 2n new_chain.batches) "batch 2 should not be stored anymore"
+    ;  Unit.assert_equals 1n new_chain.max_index "max_index did not change"
+    ] 
+
+let _test_remove_twins () =
+    let first = batch 0n 0n in
+    let second = batch 1n 0n in
+    let third = batch 1n 0n in
+    let chain = 
+    { empty_chain with
+      max_index = 1n 
+    ; batches = Big_map.literal [(1n, first) ; (2n, second) ; (3n, third)]
+    ; children = Big_map.literal [(1n, [3n;2n])]
+    } in
+    let new_chain = remove_batch (1n, chain) in
+    Unit.and_list 
+    [  Unit.assert_equals (None : batch option) (Big_map.find_opt 1n new_chain.batches) "batch 1 should not be stored anymore"
+    ;  Unit.assert_equals (None : batch option) (Big_map.find_opt 2n new_chain.batches) "batch 2 should not be stored anymore"
+    ;  Unit.assert_equals (None : batch option) (Big_map.find_opt 3n new_chain.batches) "batch 3 should not be stored anymore"
+    ;  Unit.assert_equals 1n new_chain.max_index "max_index did not change"
+    ] 
+
+let _test_remove_grand_child () =
```

Comment:
the same. the behaviour of this case is more like removing the father of grandparent

Reply:
I changed the name and description to better reflect the intent.